test/functional/test_framework/test_node.py

```diff
@@ -144,6 +144,11 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, timeout_factor, bitc
         self.p2ps = []
         self.timeout_factor = timeout_factor
 
+        # The time doesn't move forwards after we mock it, so don't call setmocktime if the test
+        # doesn't intend to use it.  If a test wants to use setmocktime, it should call
+        # bumpmocktime(0) at the beginning of run_test in order to prevent spurious failures.
+        self.mocktime = int(time.time())
```

Comment:
Sorry for being nitty about this but it's a little weird to have this set to some time that is different from the actual mock time of the node.

I would suggest initializing this to `None`, resetting it to `None` if `setmocktime(0)` was called and additionally assert that it is not None in `bumpmocktime`.

This ensures it is not set to some differing value (different from the actual mocktime) and makes sure `bumpmocktime` is used correctly. `TestNode.mocktime` can then also be reliably used as a getter for the node's mocktime.

Reply:
Done. I like that the assertion forces tests to make an initial setmocktime call before calling bumpmocktime.